src/Checker/BrokenLinkChecker.php

```diff
@@ -42,7 +49,16 @@ public function getExternalBrokenLinks(array $links): ?array
                 $status = 'invalid';
             }
 
-            $urls[$status][] = $link;
+            switch ($status) {
+                case (in_array($status, self::STATUS_CODE_REDIRECTS)):
```

Comment:
```suggestion
            switch (true) {
                case ($status >= 300 && $status < 400):
```

? (same for other ranges)

Ex in Symfony's Response:

https://github.com/symfony/symfony/blob/9e47d900191db9c6cf4ca284710b987366fffda8/src/Symfony/Component/HttpFoundation/Response.php#L1130-L1168